### ISSUE TITLE
[Bug]: Getting error FileNotFoundError: [Errno 2] No such file or directory: 'ner' while using flair ner

### ISSUE THREAD

**Issue by snamjoshi1** [flairNLP/flair#3282]
### Describe the bug

Getting below issue
File ~\Anaconda\lib\site-packages\flair\file_utils.py:50, in load_big_file(f)
     44 def load_big_file(f: str):
     45     """
     46     Workaround for loading a big pickle file. Files over 2GB cause pickle errors on certain Mac and Windows distributions.
     47     :param f:
     48     :return:
     49     """
---> 50     with open(f, "rb") as f_in:
     51         # mmap seems to be much more memory efficient
     52         bf = mmap.mmap(f_in.fileno(), 0, access=mmap.ACCESS_READ)
     53         f_in.close()

FileNotFoundError: [Errno 2] No such file or directory: 'ner'

---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[2], line 8
      5 sentence = Sentence('I love Berlin .')
      7 # load the NER tagger
----> 8 tagger = Classifier.load('ner')
     10 # run NER over sentence
     11 tagger.predict(sentence)

### To Reproduce

```python
from flair.data import Sentence
from flair.nn import Classifier

# make a sentence
sentence = Sentence('I love Berlin .')

# load the NER tagger
tagger = Classifier.load('ner')

# run NER over sentence
tagger.predict(sentence)

# print the sentence with all annotations
print(sentence)
```

### Expected behavior

should get NER results

### Logs and Stack traces

```stacktrace
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[2], line 8
      5 sentence = Sentence('I love Berlin .')
      7 # load the NER tagger
----> 8 tagger = Classifier.load('ner')
     10 # run NER over sentence
     11 tagger.predict(sentence)

File ~\Anaconda\lib\site-packages\flair\nn\model.py:559, in Classifier.load(cls, model_path)
    555 @classmethod
    556 def load(cls, model_path: Union[str, Path, Dict[str, Any]]) -> "Classifier":
    557     from typing import cast
--> 559     return cast("Classifier", super().load(model_path=model_path))

File ~\Anaconda\lib\site-packages\flair\nn\model.py:165, in Model.load(cls, model_path)
    162         continue
    164 # if the model cannot be fetched, load as a file
--> 165 state = model_path if isinstance(model_path, dict) else load_torch_state(str(model_path))
    167 # try to get model class from state
    168 cls_name = state.pop("__cls__", None)

File ~\Anaconda\lib\site-packages\flair\file_utils.py:358, in load_torch_state(model_file)
    354 warnings.filterwarnings("ignore")
    355 # load_big_file is a workaround byhttps://github.com/highway11git
    356 # to load models on some Mac/Windows setups
    357 # see https://github.com/zalandoresearch/flair/issues/351
--> 358 f = load_big_file(model_file)
    359 return torch.load(f, map_location="cpu")

File ~\Anaconda\lib\site-packages\flair\file_utils.py:50, in load_big_file(f)
     44 def load_big_file(f: str):
     45     """
     46     Workaround for loading a big pickle file. Files over 2GB cause pickle errors on certain Mac and Windows distributions.
     47     :param f:
     48     :return:
     49     """
---> 50     with open(f, "rb") as f_in:
     51         # mmap seems to be much more memory efficient
     52         bf = mmap.mmap(f_in.fileno(), 0, access=mmap.ACCESS_READ)
     53         f_in.close()

FileNotFoundError: [Errno 2] No such file or directory: 'ner'
```

### Screenshots

![image](https://github.com/flairNLP/flair/assets/40161254/9a577f5a-282f-4844-987a-d7da3505d237)

### Environment

Local

**Comment by leereak** [comment]
got the same problem, hope someone can fix it :)

**Comment by mirix** [comment]
#### Versions:
##### Flair
0.12.2
##### Pytorch
2.1.0.dev20230819+cu121
##### Transformers
4.32.0.dev0
#### GPU
True

**Comment by mirix** [comment]
If I download the model and point the Classifier to it or if I use the SequenceTagger instead, the error is different:

```
  File "/home/emoman/.local/lib/python3.11/site-packages/torch/nn/modules/module.py", line 2152, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for XLMRobertaModel:
	Unexpected key(s) in state_dict: "embeddings.position_ids".
```

I am guessing I am going to need to create a dedicated environment with PyTorch stable.

I will try that and report back.

**Comment by mirix** [comment]
Downgrading PyTorch did not help:

```
#### Versions:
##### Flair
0.12.2
##### Pytorch
2.0.1+cpu
##### Transformers
4.31.0
#### GPU
False
```

**Comment by mirix** [comment]
The solution was downgrading Transformers to v4.30.2

**Comment by CodeAKrome** [comment]
I had this too and had the versions mentioned above, no joy. I fixed it thusly (NEW virtual environment):

gh repo clone flairNLP/flair
cd flair
pip install flair
pip install -r requirements.txt

--- This then worked

from flair.nn import Classifier
from flair.data import Sentence
tagger = Classifier.load('linker')
sentence = Sentence('Kirk and Spock met on the Enterprise.')
tagger.predict(sentence)
for label in sentence.get_labels():
    print(label)

---

Without installing requirements.txt you get a semver error.


Hackers of the world, unite for great victory!

**Comment by CodeAKrome** [comment]
# Example
Run this and :boom:! Comment out any of the assignments and :smiley_cat:.

```python
from flair.nn import Classifier

tagger4 = Classifier.load('ner-large')
tagger18 = Classifier.load('ner-ontonotes-large')
link_tagger = Classifier.load('linker')
```

Upon first glance this looks like a namespace collision somehow to me. Go go Gadget debugger!

---

#### Versions:
##### Flair
0.12.2
##### Pytorch
2.0.0+cu117
##### Transformers
4.30.2
#### GPU
True

---

Hackers of the world, unite for great victory!

**Comment by CodeAKrome** [comment]
# TL;DR
Try splitting functionality. This seems to be like when the Joker poisoned products which, when used separately, were safe, but in certain combinations were lethal. It's only when I try all 3 lines things go wrong, so see if you can split out a section.

# Here's where it goes wrong where filename is apparently None.
All I did to change bad to good was comment out the last line in my example above.

```zsh
python3 -m trace -t ./bug2.py > good.txt
```

```zsh
diff --side-by-side good.txt bad.txt
```

-- snip --

```
 --- modulename: traceback, funcname: __init__                   --- modulename: traceback, funcname: __init__
traceback.py(270):         self.filename = filename             traceback.py(270):         self.filename = filename
traceback.py(271):         self.lineno = lineno                 traceback.py(271):         self.lineno = lineno
traceback.py(272):         self.name = name                     traceback.py(272):         self.name = name
traceback.py(273):         self._line = line                    traceback.py(273):         self._line = line
traceback.py(274):         if lookup_line:                      traceback.py(274):         if lookup_line:
traceback.py(276):         self.locals = {k: repr(v) for k, v   traceback.py(276):         self.locals = {k: repr(v) for k, v
traceback.py(364):         for f, lineno in frame_gen:          traceback.py(364):         for f, lineno in frame_gen:
 --- modulename: traceback, funcname: walk_stack                 --- modulename: traceback, funcname: walk_stack
traceback.py(319):         f = f.f_back                         traceback.py(319):         f = f.f_back
traceback.py(317):     while f is not None:                     traceback.py(317):     while f is not None:
traceback.py(378):         for filename in fnames:              traceback.py(378):         for filename in fnames:
traceback.py(379):             linecache.checkcache(filename)   traceback.py(379):             linecache.checkcache(filename)
 --- modulename: linecache, funcname: checkcache                 --- modulename: linecache, funcname: checkcache
linecache.py(56):     if filename is None:                      linecache.py(56):     if filename is None:
linecache.py(58):     elif filename in cache:                   linecache.py(58):     elif filename in cache:
linecache.py(61):         return                              <
traceback.py(378):         for filename in fnames:            <
traceback.py(379):             linecache.checkcache(filename) <
 --- modulename: linecache, funcname: checkcache              <
linecache.py(56):     if filename is None:                    <
linecache.py(58):     elif filename in cache:                 <
linecache.py(59):         filenames = [filename]                linecache.py(59):         filenames = [filename]
linecache.py(63):     for filename in filenames:                linecache.py(63):     for filename in filenames:
linecache.py(64):         entry = cache[filename]               linecache.py(64):         entry = cache[filename]
linecache.py(65):         if len(entry) == 1:                   linecache.py(65):         if len(entry) == 1:
linecache.py(68):         size, mtime, lines, fullname = entr   linecache.py(68):         size, mtime, lines, fullname = entr
linecache.py(69):         if mtime is None:                     linecache.py(69):         if mtime is None:
linecache.py(71):         try:                                  linecache.py(71):         try:
linecache.py(72):             stat = os.stat(fullname)          linecache.py(72):             stat = os.stat(fullname)
linecache.py(76):         if size != stat.st_size or mtime !=   linecache.py(76):         if size != stat.st_size or mtime !=
linecache.py(63):     for filename in filenames:                linecache.py(63):     for filename in filenames:
traceback.py(378):         for filename in fnames:              traceback.py(378):         for filename in fnames:
traceback.py(379):             linecache.checkcache(filename)   traceback.py(379):             linecache.checkcache(filename)
 --- modulename: linecache, funcname: checkcache                 --- modulename: linecache, funcname: checkcache
linecache.py(56):     if filename is None:                      linecache.py(56):     if filename is None:
linecache.py(58):     elif filename in cache:                   linecache.py(58):     elif filename in cache:
                                                              > linecache.py(61):         return
                                                              > traceback.py(378):         for filename in fnames:
                                                              > traceback.py(379):             linecache.checkcache(filename)
                                                              >  --- modulename: linecache, funcname: checkcache
                                                              > linecache.py(56):     if filename is None:
                                                              > linecache.py(58):     elif filename in cache:
linecache.py(59):         filenames = [filename]                linecache.py(59):         filenames = [filename]
linecache.py(63):     for filename in filenames:                linecache.py(63):     for filename in filenames:
linecache.py(64):         entry = cache[filename]               linecache.py(64):         entry = cache[filename]
linecache.py(65):         if len(entry) == 1:                   linecache.py(65):         if len(entry) == 1:
linecache.py(67):             continue                          linecache.py(67):             continue
linecache.py(63):     for filename in filenames:                linecache.py(63):     for filename in filenames:
```

**Comment by RamesanPP** [comment]
Transformers v4.30.2 works for me as well

**Comment by beljul** [comment]
Hello ! We still have the issue with transformers >= 4.31.0. Because of that we have conflicts with recent protobuf version because transformers (4.30.2) depends on protobuf (<=3.20.3).

Do you have another fix than downgrading the transformers version ?

**Comment by mirix** [comment]
This bug is not fixed yet. This forces us to have to use flair in a dedicated environment with an old version of Transformers.
